st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -277,17 +294,35 @@ def validate(self):
 
             return self
 
-    @jsexpose(body_cls=ExecutionSpecificationAPI, status_code=http_client.CREATED)
-    def post(self, spec_api, execution_id, no_merge=False):
+    def post(self, spec_api, id, requester_user=None, no_merge=False, **kwargs):
         """
         Re-run the provided action execution optionally specifying override parameters.
 
         Handles requests:
 
             POST /executions/<id>/re_run
         """
+
+        if (spec_api.tasks or spec_api.reset) and spec_api.parameters:
```

Comment:
I believe we've already discussed and even worked on making `st2-register-content` work through the API rather than accessing DB directly. Yep, we're not there yet and we would need to figure out how to minimize the risk in the mean time. But in the long run, it seems to me a right way to go.